idaes/models_extra/power_generation/unit_models/soc_submodels.py

```diff
@@ -0,0 +1,3997 @@
+class SocChannelData(UnitModelBlockData):
+    CONFIG = ConfigBlock()
+    CONFIG.declare(
+        "dynamic",
+        ConfigValue(
+            domain=In([useDefault, True, False]),
+            default=useDefault,
+            description="Dynamic model flag",
+            doc="""Indicates whether this model will be dynamic,
+                **default** = useDefault.
+                **Valid values:** {
+                **useDefault** - get flag from parent (default = False),
+                **True** - set as a dynamic model,
+                **False** - set as a steady-state model.}""",
+        ),
+    )
+    CONFIG.declare(
+        "has_holdup",
+        ConfigValue(domain=In([useDefault, True, False]), default=useDefault),
+    )
+    CONFIG.declare(
+        "comp_list",
+        ConfigValue(default=["H2", "H2O"], description="List of components"),
+    )
+    
+    CONFIG.declare(
+        "cv_zfaces",
+        ConfigValue(description="CV boundary set, should start with 0 and end with 1."),
+    )
+    CONFIG.declare(
+        "length_z",
+        ConfigValue(
+            default=None, description="Length in the direction of flow (z-direction)"
+        ),
+    )
+    CONFIG.declare(
+        "length_y", ConfigValue(default=None, description="Width of cell (y-direction)")
+    )
+    CONFIG.declare(
+        "include_temperature_x_thermo",
+        ConfigValue(domain=In([useDefault, True, False]),
+                    default=True,
+                    description="Whether to consider temperature variations in "
+                    "x direction in thermodynamic equations"),
+    )
+    _thermal_boundary_conditions_config(CONFIG,thin=False)
+    CONFIG.declare(
+        "opposite_flow",
+        ConfigValue(default=False, description="If True assume velocity is negative"),
+    )
+    CONFIG.declare(
+        "below_electrode",
+        ConfigValue(domain=In([True, False]),
+                    description="Decides whether or not to create material "
+                    "flux terms above or below the channel."),
+    )
+    CONFIG.declare(
+        "interpolation_scheme",
+        ConfigValue(
+            default=CV_Interpolation.UDS,
+            description="Method used to interpolate face values",
+        ),
+    )
+
+    def build(self):
+        super().build()
+
+        # Set up some sets for the space and time indexing
+        dynamic = self.config.dynamic
+        tset = self.flowsheet().config.time
+        self.comps = pyo.Set(initialize=self.config.comp_list)
+        # z coordinates for nodes and faces
+        self.zfaces = pyo.Set(initialize=self.config.cv_zfaces)
+        self.znodes = pyo.Set(
+            initialize=[
+                (self.zfaces.at(i) + self.zfaces.at(i + 1)) / 2.0
+                for i in range(1, len(self.zfaces))
+            ]
+        )
+        # This sets provide an integer index for nodes and faces
+        self.izfaces = pyo.Set(initialize=range(1, len(self.zfaces) + 1))
```

Comment:
Do you need these? I think there are Pyomo methods for Sets that would let you achieve this using the existing `Sets`.

Reply:
pyo.RangeSet()?

Reply:
I was thinking `pyo.Set.at()` and the like.

Reply:
The big question to me is "how do we want users to reference variables in the model?" Initially I was going to change things to use Pyomo's idiosyncratic "index continuum variables by floats" method in order to be consistent with what's implemented in `ControlVolume1D`. However, the only discretization methods supported are finite differences and collocation, in which equations are written for the points at `zfaces` (as well as additional, collocation points for collocation). However, a finite volume method is used here, and `znodes` are centered at the midpoint between the points specified in the faces. I don't know that users, having passed `zfaces = [0, 0.25, 0.5, 0.75, 1]` will be thrilled having to index `temperature_z` using `znodes=[0.125, 0.375, 0.625, 0.875]`.